dor/providers/descriptor_generator.py

```diff
@@ -1,42 +1,37 @@
 from pathlib import Path
 from datetime import datetime, UTC
 
-from dor.settings import template_env
 from dor.providers.models import PackageResource
+from dor.settings import template_env
 
-def relative_path_or_not(locref: str):
-    if locref.startswith("https://"):
-        return locref
-    return f"../{locref}"
 
-def build_descriptor_filename(resource):
-    return f"{resource.id}.{resource.type.lower().replace(" ", "_")}.mets2.xml"
+def build_descriptor_file_path(resource):
+    return Path(str(resource.id)) / "descriptor" / f"{resource.id}.{resource.type.lower().replace(" ", "_")}.mets2.xml"
 
 class DescriptorGenerator:
     def __init__(self, package_path: Path, resources: list[PackageResource]):
         self.package_path = package_path
         self.resources = resources
+
         self.entries = []
 
     def write_files(self):
         struct_map_locref_data = {}
         for resource in self.resources:
-            if resource.type == "Asset":
+            if resource.type == "File Set":
                 identifier = f"urn:dor:{resource.id}"
-                struct_map_locref_data[identifier] = build_descriptor_filename(resource)
+                struct_map_locref_data[identifier] = build_descriptor_file_path(resource)
 
         entity_template = template_env.get_template("preservation_mets.xml")
         for resource in self.resources:
             xmldata = entity_template.render(
-                relative_path_or_not=relative_path_or_not,
                 resource=resource,
                 struct_map_locref_data=struct_map_locref_data,
-                action="stored",
                 create_date=datetime.now(tz=UTC).strftime("%Y-%m-%dT%H:%M:%SZ"),
             )
-            filename = Path(f"{resource.id}/descriptor/{build_descriptor_filename(resource)}")
-            output_filename = self.package_path / filename
-            output_filename.parent.mkdir(parents=True, exist_ok=True)
-            with (output_filename).open("w") as f:
+            descriptor_file_path = build_descriptor_file_path(resource)
+            output_path = self.package_path / descriptor_file_path
+            output_path.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
I thought about (and tried) incorporating `FilesystemFileProvider` here, but really this is only needed for the unit tests (hence the `exist_ok`, which makes it work when we're replacing existing files). `FilesystemFileProvider.create_directories` doesn't yet support `exist_ok`. Not exactly sure what to do here, but I was wary of adding too many more changes to the PR.